### PR TITLE
Jwilkes nmap 08232022

### PR DESCRIPTION
**Pull request by johnnywilkes**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
We were creating playbook for NMAP scans for a project and decided to make them modular enough to share with all.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [ ] Tests
- [X] Documentation

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@ShacharKidor , @bakatzir said to work with you on PRs.  This is the first of many so feedback is welcome!

CC: @andrew-paloalto

File: Packs/Nmap/ReleaseNotes/1_1_0.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+#### Playbooks
+##### New: NMAP - Single Port Scan
+- Added the "NMAP - Single Port Scan" playbook to conduct a single port NMAP scan.
```
Comment (ShacharKidor) [inline Packs/Nmap/ReleaseNotes/1_1_0.md:4]:
```suggestion
- Added the **NMAP - Single Port Scan** playbook to conduct a single port NMAP scan.
```

File: Packs/Nmap/ReleaseNotes/1_1_0.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+##### New: NMAP - Banner Check
+- Added the "NMAP - Banner Check" to conduct a single port NMAP scan and check regular expression against banner.
```
Comment (ShacharKidor) [inline Packs/Nmap/ReleaseNotes/1_1_0.md:7]:
```suggestion
- Added the **NMAP - Banner Check"** to conduct a single port NMAP scan and check regular expression against the banner.
```

File: Packs/Nmap/pack_metadata.json [hunk 3]
```diff
@@ -16,5 +16,14 @@
...
         "marketplacev2"
+    ],
+    "dependencies": {
+        "CommonScripts": {
+            "mandatory": true,
+            "display_name": "Common Scripts"
+        }
+    },
+    "displayedImages": [
+        "CommonScripts"
```
Comment (ShacharKidor) [inline Packs/Nmap/pack_metadata.json:27]: Please remove that. 
```suggestion
```

**Review by ShacharKidor — CHANGES_REQUESTED** [review]
Hi @johnnywilkes, 
Thank you for your contribution! looks great. 

I added 3 little fixes to the Release note and pack metadata files - please commit them. 

@idovandijk - Please review the 2 new playbooks. 

I'm available here or via slack for any questions. 

Thanks again :)

**Comment by johnnywilkes** [comment]
@ShacharKidor, thank you for the quick review!  I have commited your suggestions.

**Comment by ShirleyDenkberg** [comment]
@johnnywilkes I created a pack readme for the NMAP content pack. I sent it to you via Slack. Please add to this PR. This content will appear in Marketplace. Feel free to edit the content if you want.
@ShacharKidor @idovandijk Doc review completed.

**Comment by johnnywilkes** [comment]
@ShirleyDenkberg , I have added the README file

**Review by idovandijk — CHANGES_REQUESTED** [review]
Hi @johnnywilkes, thank you for this contribution. I really like your playbooks and have to say your work is very clean, so there is not much to change here. 

I do have to point out what I think is a logical flaw.
It looks like in the banner matching playbook you output the same results as in the port scan, but what is interesting is whether the banner matched the regex.
image

Also a suggestion - since both subplaybooks check different things, I would suggest that you change the output to be something like `NmapPortScanResult` and `NmapBannerCheckResult` so if both are used as subplaybooks, their results would still be distinguishable.

If you think I am mistaken please let me know why. Otherwise please make the changes/fixes and let me know once you're done.
Thanks again!

**Comment by johnnywilkes** [comment]
@idovandijk , thanks for looking this over.  the reason for these playbooks for us was to determine if a ASM issue was still present.  This could be something like:
- Is RDP port open?
- Is SSH port open and banner matches insecure OpenSSH version?

Therefore, I thought the the result should be either: yes the issue is still observable, no the issue isn't observable or inconclusive.

I don't mind changing `ScanResult` to `NmapPortScanResult` or `NmapBannerCheckResult` but that would also result in more logic added to its parent playbook.

**Comment by idovandijk** [comment]
> 

I see, that makes more sense now. If the use-case is as you said then it would make sense to keep the current outputs I believe. However, I don't see the parent playbook as part of this PR. Were you planning to contribute that playbook? Seeing it could give better context of how these 2 are meant to be used in your case. I'm not sure if it's clear enough for the banner matching playbook (simply returns Open if the regex matches)

**Comment by johnnywilkes** [comment]
@idovandijk , because these playbooks are modular we talked to @bakatzir about adding the Nmap content back.  The parent is less modular and more specific to ASM so we will add the the ASM content pack at a later time.  I thought I had some decent explanation:
image
image

Any suggested changes to make it more clear?

**Comment by idovandijk** [comment]
> @idovandijk , because these playbooks are modular we talked to @bakatzir about adding the Nmap content back. The parent is less modular and more specific to ASM so we will add the the ASM content pack at a later time. I thought I had some decent explanation: image image
> 
> Any suggested changes to make it more clear?

I missed that description, thanks :)

**Review by idovandijk — APPROVED** [review]
Great job

**Review by ShacharKidor — APPROVED** [review]
@johnnywilkes  Great Job!

**Pull request merged**